### PR TITLE
Calender UI v1

### LINKED ISSUE #178: Calender UI + API Integration (partial)
Add Calender UI card and use getEvents API

### PR DESCRIPTION
**Pull request by SriramPatibanda**
# Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

## Description

Fixes #178 

 - Add LinerProgressIndicator for WebView
 - Add GraphQL Service
 - Add Event Model 
 - Add Event Repository
 - Fetch Events (Mock Data)
 - Upgrade Dependencies

💖Thank you!

## Post merge checklist

- [x] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [x] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: elaichi/lib/data/remote/graphql/graphql_service.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  GraphQLService._internal();
+
+  late final GraphQLClient _client;
+  static final GraphQLService _graphQLService = GraphQLService._internal();
+
+  void init() {
+    final _httpLink = HttpLink(
+      'https://test.dscnitrourkela.org/graphql',
```
Comment (abhibhaw) [inline elaichi/lib/data/remote/graphql/graphql_service.dart:17]: R we planning to move it to remote config?

File: elaichi/lib/domain/models/event.freezed.dart [hunk 2]
```diff
@@ -0,0 +1,316 @@
+// coverage:ignore-file
+// GENERATED CODE - DO NOT MODIFY BY HAND
+// ignore_for_file: type=lint
+// ignore_for_file: unused_element, deprecated_member_use, deprecated_member_use_from_same_package, use_function_type_syntax_for_parameters, unnecessary_const, avoid_init_to_null, invalid_override_different_default_values_named, prefer_expression_function_bodies, annotate_overrides, invalid_annotation_target
+
+part of 'event.dart';
+
```
Comment (abhibhaw) [inline elaichi/lib/domain/models/event.freezed.dart:7]: Shouldn't we keep this in gitignore?

File: elaichi/lib/domain/models/event.g.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+part of 'event.dart';
+
```
Comment (abhibhaw) [inline elaichi/lib/domain/models/event.g.dart:4]: same!!

File: elaichi/lib/presentation/profile/bloc/profile_bloc.dart [hunk 4]
```diff
@@ -38,4 +44,13 @@ class ProfileBloc extends Bloc<ProfileEvent, ProfileState> {
       return false;
     }
   }
+
+  List<Event> getEvents() {
+    return _eventRepository.events!;
+  }
+
+  Future<List<Event>?> refreshAndGetEvents() async {
+    await _eventRepository.fetchEvents();
```
Comment (abhibhaw) [inline elaichi/lib/presentation/profile/bloc/profile_bloc.dart:53]: You're rethrowing the error in fetchEvents, shouldn't it should be catched again here?

File: elaichi/lib/presentation/profile/bloc/profile_bloc.dart [hunk 5]
```diff
@@ -38,4 +44,13 @@ class ProfileBloc extends Bloc<ProfileEvent, ProfileState> {
...
+    return _eventRepository.events;
```
Comment (abhibhaw) [inline elaichi/lib/presentation/profile/bloc/profile_bloc.dart:54]: Let's assume that the fetch event API call failed for some reason. Won't it throw late initialisation error here?

File: elaichi/lib/presentation/core/app.dart [hunk 6]
```diff
@@ -27,14 +29,23 @@ class ElaichiApp extends StatelessWidget {
   static Future<Widget> run() async {
     final localStorageService = await LocalStorageService.init();
     final apiService = APIService();
+    final graphQLService = GraphQLService()..init();
```
Comment (abhibhaw) [inline elaichi/lib/presentation/core/app.dart:32]: is this .. a type? or some syntax I'm not aware of?

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: elaichi/lib/domain/models/event.g.dart (hunk 3 above)
Comment (SriramPatibanda) [reply elaichi/lib/domain/models/event.g.dart:4]: No need to do that for models and freezed packages. I want them in the repo.

File: elaichi/lib/presentation/profile/bloc/profile_bloc.dart (hunk 4 above)
Comment (SriramPatibanda) [reply elaichi/lib/presentation/profile/bloc/profile_bloc.dart:53]: Refresh and fetch events was a function I wrote for testing purposes. I'm not using it anywhere as of now.

File: elaichi/lib/presentation/core/app.dart (hunk 6 above)
Comment (SriramPatibanda) [reply elaichi/lib/presentation/core/app.dart:32]: It's a syntax that you don't know.

File: elaichi/lib/data/remote/graphql/graphql_service.dart (hunk 1 above)
Comment (SriramPatibanda) [reply elaichi/lib/data/remote/graphql/graphql_service.dart:17]: Let it be here as of now. I'm yet to decide how I would like to manage such links as I'll be using mm ke graphql also

File: elaichi/lib/presentation/core/app.dart (hunk 6 above)
Comment (abhibhaw) [reply elaichi/lib/presentation/core/app.dart:32]: Pls help me know a bit more abt this, any docs will be helpful :)

File: elaichi/lib/data/remote/graphql/graphql_service.dart (hunk 1 above)
Comment (riteshsp2000) [reply elaichi/lib/data/remote/graphql/graphql_service.dart:17]: Let's try to manage things appropriately from the start. Refactoring it later will be a huge task. Ideally we shouldn't be going ahead with this.

File: elaichi/lib/presentation/profile/bloc/profile_bloc.dart (hunk 4 above)
Comment (riteshsp2000) [reply elaichi/lib/presentation/profile/bloc/profile_bloc.dart:53]: If you are not using this then please remove it. If you plan to use it in future then add some sort of comment/indication that it isn't in use at the moment, here for furture purposes.

File: elaichi/lib/data/remote/graphql/graphql_service.dart (hunk 1 above)
Comment (SriramPatibanda) [reply elaichi/lib/data/remote/graphql/graphql_service.dart:17]: Yep, I'll make changes accordingly in the next PR with remote config service and MM integration

Comment (SriramPatibanda) [reply elaichi/lib/data/remote/graphql/graphql_service.dart:17]: and no, refactoring this specific thing isn't going to be a huge task. It's pretty much decoupled from any service.

File: elaichi/lib/domain/models/event.g.dart (hunk 3 above)
Comment (riteshsp2000) [reply elaichi/lib/domain/models/event.g.dart:4]: Not sure what the appropriate way is, can someone please elaborate a bit here?

File: elaichi/lib/presentation/profile/bloc/profile_bloc.dart (hunk 4 above)
Comment (SriramPatibanda) [reply elaichi/lib/presentation/profile/bloc/profile_bloc.dart:53]: I tend to reuse it whenever I want and there's no collaborator as of now. I would want it to be here, I don't see any problem.

File: elaichi/lib/domain/models/event.g.dart (hunk 3 above)
Comment (SriramPatibanda) [reply elaichi/lib/domain/models/event.g.dart:4]: https://www.reddit.com/r/FlutterDev/comments/kazxo0/do_you_add_gdart_files_to_gitignore/

it's mostly a preference.

**Review by abhibhaw — APPROVED** [review]

**Pull request merged**